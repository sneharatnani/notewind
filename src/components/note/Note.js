import { useState } from "react";
import NotePreview from "../NotePreview.js";
import NoteLabel from "./NoteLabel.js";

export default function Note(props) {
  const { title, body, label, bg } = props;
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <div className="w-[calc(100% / 1)] mb-4">
        <div
          data-testid="note"
          className={`group w-full rounded-lg p-4 border border-gray-200 box-border overflow-hidden cursor-pointer hover:shadow-sm hover:border-gray-300 flex flex-col justify-between ${bg}`}
          onClick={() => {
            setIsOpen(true);
          }}
        >
          <div>
            {" "}
            <h3 className="font-medium text-lg">{title}</h3>
            <p className="text-sm font-poppins break-words whitespace-pre-wrap overflow-hidden max-h-60">
              {body}
            </p>
          </div>
          {label.length !== 0 && (
            <div className="mt-2">
              {[...label].map((l) => (
                <NoteLabel label={l} key={l} />
              ))}
            </div>
          )}
        </div>
      </div>
      <NotePreview {...props} isOpen={isOpen} setIsOpen={setIsOpen} />
    </>
  );
}
