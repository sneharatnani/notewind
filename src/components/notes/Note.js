export default function Note(props) {
  const { title, body } = props;
  function findNewlines() {
    const newLines = body.split(/\n/).join("<br/>");
    return newLines;
  }

  return (
    <div
      className="w-full rounded-lg p-4 border border-gray-200 max-h-60 overflow-hidden box-border"
      onClick={findNewlines}
    >
      <p className="font-medium text-lg">{title}</p>
      <pre className="text-sm whitespace-pre-wrap break-words font-poppins">
        {body}
      </pre>
    </div>
  );
}
