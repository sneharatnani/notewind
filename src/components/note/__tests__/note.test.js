import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import Note from "../Note.js";

jest.mock("../../NotePreview.js");
jest.mock("./../NoteLabel.js", () => ({ label }) => {
  return <span>{label}</span>;
});

describe("Note", () => {
  it("should show props correctly", () => {
    const { container } = render(
      <Note title="title" body="note body" label={["label"]} bg="bg-white" />
    );

    expect(container).toMatchSnapshot();
  });

  it("calls setIsOpen", () => {
    const setState = jest.fn();
    const useStateMock = (initState) => [initState, setState];
    jest.spyOn(React, "useState").mockImplementation(useStateMock);

    render(
      <Note title="title" body="note body" label={["label"]} bg="bg-white" />
    );
    userEvent.click(screen.getByTestId("note"));

    expect(setState).toHaveBeenCalledTimes(1);
  });
});
