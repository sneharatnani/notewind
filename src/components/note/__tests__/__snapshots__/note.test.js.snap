// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note should show props correctly 1`] = `
<div>
  <div
    class="w-[calc(100% / 1)] mb-4"
  >
    <div
      class="group w-full rounded-lg p-4 border border-gray-200 box-border overflow-hidden cursor-pointer hover:shadow-sm hover:border-gray-300 flex flex-col justify-between bg-white"
      data-testid="note"
    >
      <div>
         
        <h3
          class="font-medium text-lg"
        >
          title
        </h3>
        <p
          class="text-sm font-poppins break-words whitespace-pre-wrap overflow-hidden max-h-60"
        >
          note body
        </p>
      </div>
      <div
        class="mt-2"
      >
        <span>
          label
        </span>
      </div>
    </div>
  </div>
</div>
`;
